examples/TaskUsage/main.roc

```diff
@@ -0,0 +1,100 @@
+app "task-usage"
+    packages { pf: "https://github.com/roc-lang/basic-cli/releases/download/0.3.2/tE4xS_zLdmmxmHwHih9kHWQ7fsXtJr7W7h3425-eZFk.tar.br" }
+    imports [
+        pf.Stdout,
+        pf.Stderr,
+        pf.Task,
+    ]
+    provides [main] to pf
+
+## A Task represents an effect; an interaction with state outside your Roc program, such as the terminal's standard output, or a file.
+
+## A simple Task that always succeeds.
+##
+## A Task has two type parameters: the type of value it produces when it finishes running,
+## and any errors that might happen when running it.
+##
+## The type `Task Str []` indicates that the success value of this task is
+## `Str` and it cannot fail, so the error type is empty. 
+##
+getName : Task.Task Str []
+getName = Task.succeed "Louis"
+
+## This task uses the `getName` task and combines the returned `Str` with the
+## welcome message "Bonjour".
+##
+## We use `Task.map` to take the success value of the `getName` task, a `Str` and apply a function
+## to it with type `Str -> Str`: `\name -> "Bonjour \(name)!"`.
+##
+## The type of `Task.map` is `Task a err, (a -> b) -> Task b err`.
+## Filling this in for our case, this becomes: `Task Str [], (Str -> Str) -> Task Str []`
+## In this case the type variables `a` and `b` are both `Str`, the types may be different but
+## they don't need to be.
+##
+## Remember that `Task.map` is used to transform (= apply a function to) the success value of a Task.
+##
+## Next we use `Task.await` to print the new string to `Stdout`. This
+## function takes the `Str` and returns a new `Task`.
+##
+## `Stdout.line` has the type `Str -> Task {} *`.
```

Comment:
@Anton-4 I think there is no difference between `Task Str []` and `Task Str *`, I'll ask in zulip.